lib/Strategy/MixinStrategy.php

```diff
@@ -43,7 +44,12 @@ public function createNodeForDocument($document, NodeInterface $parentNode, $nam
 
         $node = $parentNode->addNode($name);
         $node->addMixin($metadata->getPhpcrType());
-        $node->setProperty('jcr:uuid', UUIDHelper::generateUUID());
+
+        if (!$document instanceof UuidBehavior || !$uuid = $document->getUuid()) {
+            $uuid = UUIDHelper::generateUUID();
```

Comment:
Add comment about what this does: it will keep existing UUIDs set on new documents (which may happen when persisting a document from one DM to another).